src/Interpreters/Cache/Metadata.cpp

```diff
@@ -53,11 +53,17 @@ KeyMetadata::KeyMetadata(
     const Key & key_,
     const std::string & key_path_,
     CleanupQueue & cleanup_queue_,
+    [[maybe_unused]] Poco::Logger * log_,
     bool created_base_directory_)
     : key(key_)
     , key_path(key_path_)
     , cleanup_queue(cleanup_queue_)
     , created_base_directory(created_base_directory_)
+#ifdef ABORT_ON_LOGICAL_ERROR
+    , log(&Poco::Logger::get("Key(" + key.toString() + ")"))
```

Comment:
Let's remove it entirely. We should not put a high-cardinality value into the logger name.

Reply:
Just put the key name into the messages.

Reply:
I don't know, it looks ok to me: only debug and sanitizer builds.

Reply:
But it's not intended to put a high-cardinality string into the logger name.

Reply:
Removed.